server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -423,7 +438,7 @@ private static boolean updateCustomRuleFromTemplateRule(RuleDto customRule, Rule
   private void removeActiveRulesOnStillExistingRepositories(DbSession session, Collection<RuleDto> removedRules, RulesDefinition.Context context) {
     List<String> repositoryKeys = newArrayList(Iterables.transform(context.repositories(), new Function<RulesDefinition.Repository, String>() {
       @Override
-      public String apply(RulesDefinition.Repository input) {
+      public String apply(@Nonnull RulesDefinition.Repository input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The "input" parameter nullability is different in the superclass method, and that should not be changed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)